Patch from Martin Michlmayr for building with GCC 4.4.

Index: libitpp/itpp/base/timing.cpp
===================================================================
--- libitpp.orig/itpp/base/timing.cpp	2009-04-26 17:15:37.000000000 -0500
+++ libitpp/itpp/base/timing.cpp	2009-04-26 17:15:54.000000000 -0500
@@ -45,6 +45,7 @@
 #endif
 
 #include <itpp/base/timing.h>
+#include <cstdio>
 #include <iostream>
 #include <cmath>
 
Index: libitpp/itpp/base/vec.cpp
===================================================================
--- libitpp.orig/itpp/base/vec.cpp	2009-04-26 17:16:07.000000000 -0500
+++ libitpp/itpp/base/vec.cpp	2009-04-26 17:16:31.000000000 -0500
@@ -29,6 +29,7 @@
 
 #include <itpp/base/vec.h>
 #include <itpp/base/converters.h>
+#include <cstdio>
 #include <limits>
 
 //! \cond
Index: libitpp/itpp/fixed/cfix.cpp
===================================================================
--- libitpp.orig/itpp/fixed/cfix.cpp	2009-04-26 17:16:43.000000000 -0500
+++ libitpp/itpp/fixed/cfix.cpp	2009-04-26 17:16:52.000000000 -0500
@@ -29,6 +29,7 @@
 
 #include <itpp/fixed/cfix.h>
 #include <itpp/base/itassert.h>
+#include <cstdio>
 #include <iostream>
 
 
Index: libitpp/itpp/fixed/fix.cpp
===================================================================
--- libitpp.orig/itpp/fixed/fix.cpp	2009-04-26 17:16:58.000000000 -0500
+++ libitpp/itpp/fixed/fix.cpp	2009-04-26 17:17:07.000000000 -0500
@@ -29,6 +29,7 @@
 
 #include <itpp/fixed/fix.h>
 #include <itpp/base/itassert.h>
+#include <cstdio>
 #include <iostream>
 
 
